client/v3/client.go

```diff
@@ -235,13 +189,12 @@ func (c *Client) dialSetupOpts(creds grpccredentials.TransportCredentials, dopts
 	}
 	opts = append(opts, dopts...)
 
-	dialer := endpoint.Dialer
 	if creds != nil {
 		opts = append(opts, grpc.WithTransportCredentials(creds))
 	} else {
 		opts = append(opts, grpc.WithInsecure())
 	}
-	opts = append(opts, grpc.WithContextDialer(dialer))
+	grpc.WithDisableRetry()
```

Comment:
Can we add a comment on this? So, now gRPC enables retry by default?

Reply:
> So, now gRPC enables retry by default?

I don't think so.

```
	// Retry is set if retry is explicitly enabled via "GRPC_GO_RETRY=on".
	Retry = strings.EqualFold(os.Getenv(retryStr), "on")
```

Reply:
I removed this in follow up PR: 

https://github.com/etcd-io/etcd/pull/12706/commits/4a1c24556c22f55e75cb5167d555a8c367ad4d2b#diff-2bbaa3c83234888775aabf7697076e8035451045c479938e03ebb5c929132679L209

In general we should replace clientv3 retry interceptors with RetryPolicy.